Cargo.toml

```diff
@@ -13,12 +13,17 @@ keywords = ["checksum", "crc", "crc32", "simd", "fast"]
 
 [dependencies]
 cfg-if = "0.1"
+raw-cpuid = { version = "6.1", optional = true }
 
 [dev-dependencies]
 bencher = "0.1"
 quickcheck = { version = "0.6", default-features = false }
 rand = "0.4"
 
+[features]
+default = []
+no_std = ["raw-cpuid"]
```

Comment:
The general API guideline is that features should be additive. Pretty much every official / semi-official rust-lang / rust-lang-nursery library uses `use_std` as the feature, which one can enable by default.

Reply:
I know that, but mostly because they have more dependencies when `std` is enabled, which is not the case here. Do you know the proper way to have a `use_std` optional feature, enable by default, which when disabled makes the `raw_cpuid` a requirement ?

Reply:
> Do you know the proper way to have a use_std optional feature, enable by default, which when disabled makes the raw_cpuid a requirement ?

Something like

```toml
[target.`cfg(not(feature = "use_std"))`.dependencies.raw_cpuid]
version = "6.1"
```

?

Reply:
Damn, giving these a try it did not work, and I ended up landing on the same cargo bug that you mentioned 😆